var/spack/repos/builtin/packages/ccache/package.py

```diff
@@ -27,9 +31,27 @@ class Ccache(AutotoolsPackage):
     version('3.3',   sha256='b220fce435fe3d86b8b90097e986a17f6c1f971e0841283dd816adb238c5fd6a')
     version('3.2.9', sha256='1e13961b83a3d215c4013469c149414a79312a22d3c7bf9f946abac9ee33e63f')
 
-    depends_on('gperf')
-    depends_on('libxslt')
-    depends_on('zlib')
+    depends_on('zstd', when='@4.0:')
+
+    depends_on('gperf', when='@:3.99')
+    depends_on('libxslt', when='@:3.99')
+    depends_on('zlib', when='@:3.99')
+
+    # Before 4.0 this was an Autotools package
+    @when('@:3.99')
+    def cmake(self, spec, prefix):
+        pass
+
+    @when('@:3.99')
+    def build(self, spec, prefix):
+        pass
+
+    @when('@:3.99')
+    def install(self, spec, prefix):
+        configure_args = ["--prefix=" + prefix]
+        configure(*configure_args)
+        make()
+        make("install")
```

Comment:
Could these steps not be mapped better to be more fault tolerant?
I.e., `configure(…)` goes into `cmake`, `make()` into `build`? I've had issues with LLVM variants where the `install` step does actual work that failed, making it a lot harder to debug the underlying cause, and actually leading to an incomplete installation that is not flagged as a failed build.

Reply:
Thanks for the idea, I just did that :upside_down_face: